dolphin/task_manager/tasks/task.py

```diff
@@ -55,3 +55,78 @@ def sync(self):
 
     def remove(self):
         pass
+
+
+class StoragePoolTask(StorageResourceTask):
+    def __init__(self, context, storage_id):
+        super(StoragePoolTask, self).__init__(context, storage_id)
+
+    def _get_delete_list(self, storage_pools, db_pools):
+        """
+        :param storage_pools:
+        :param db_pools:
+        :return: prepare the delete_list. It will contain only those pool id
+        which is present in db but NOT in current list of storage pools
+        """
+        delete_list = copy.copy(db_pools)
+        [delete_list.remove(d_pool) for s_pool in storage_pools
+         for d_pool in db_pools if s_pool.get('original_id') == d_pool.get(
+            'original_id')]
+        return delete_list
+
+    def _get_add_list(self, storage_pools, db_pools):
+        """
+        :param storage_pools:
+        :param db_pools:
+        :return: prepare the add_list. It will contain only those pool id
+        which is NOT present in db but in current list of storage pools
+        """
+        add_list = copy.copy(storage_pools)
+        [add_list.remove(s_pool) for s_pool in storage_pools
+         for d_pool in db_pools if s_pool.get('original_id') == d_pool.get(
+            'original_id')]
+        return add_list
+
+    def _get_update_list(self, storage_pools, db_pools):
```

Comment:
Suggest to get delete_list, add_list and update_list in one function, for efficiency reason.

Reply:
how about the below logic:
```
def _classify_pools(self, storage_pools, db_pools):
    original_ids_in_db = [e['original_id'] for e in db_pools]
    delete_id_list = [e['id'] for e in db_pools]
    add_list = []
    update_list = []
    
    for x in storage_pools:
        if x['original_id'] in original_ids_in_db:
            x['id'] = db_pools[original_ids_in_db.index(
                x['original_id'])]['id']
            delete_id_list.pop(x['id'])
            update_list.append(x)
        else:
            add_list.append(x)
    
    return add_list, update_list, delete_id_list
```

Reply:
done